src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/showConfirmationPaymentFromComponent/payment.js

```diff
@@ -89,29 +91,77 @@ function handleAuthorisedPayment(
   return next();
 }
 
+function handlePaymentResult(result, order, adyenPaymentInstrument, options) {
+  // Authorised: The payment authorisation was successfully completed.
+  if (
+    [
+      constants.RESULTCODES.AUTHORISED,
+      constants.RESULTCODES.PENDING,
+      constants.RESULTCODES.RECEIVED,
+    ].indexOf(result.resultCode) > -1
+  ) {
+    return handleAuthorisedPayment(
+      order,
+      result,
+      adyenPaymentInstrument,
+      options,
+    );
+  }
+  return handlePaymentError(order, adyenPaymentInstrument, options);
+}
+
+// Check result for Apple Pay, does not require /payments/details
+function checkApplePayResponse(adyenPaymentInstrument, result) {
+  const paymentMethodVariant =
+    adyenPaymentInstrument.custom.adyenAction?.additionalData
+      ?.paymentMethodVariant;
+  if (
+    AdyenHelper.isApplePay(paymentMethodVariant) &&
+    result?.fullResponse?.isFinal
+  ) {
+    return JSON.parse(adyenPaymentInstrument.custom.adyenAction);
+  }
+  return false;
+}
+
 // eslint-disable-next-line complexity
 function handlePayment(stateData, order, options) {
   const paymentInstruments = order.getPaymentInstruments(
     constants.METHOD_ADYEN_COMPONENT,
   );
-  const result = options.req.form?.result;
+  const result = JSON.parse(options.req.form?.result);
   const adyenPaymentInstrument = paymentInstruments[0];
   const hasStateData = stateData?.paymentData && stateData?.details;
 
+  if (result.error) {
+    return handlePaymentError(order, adyenPaymentInstrument, options);
+  }
+
   let finalResult;
   if (!hasStateData) {
+    const applePayResponse = checkApplePayResponse(
+      adyenPaymentInstrument,
+      result,
+    );
     if (result && JSON.stringify(result).indexOf('amazonpay') > -1) {
       finalResult = JSON.parse(result);
+    } else if (applePayResponse) {
+      return handlePaymentResult(
+        applePayResponse,
+        order,
+        adyenPaymentInstrument,
+        options,
+      );
     } else {
-      return handlePaymentError(order, options);
+      return handlePaymentError(order, adyenPaymentInstrument, options);
     }
   }
 
   if (order.status.value === Order.ORDER_STATUS_FAILED) {
     Logger.getLogger('Adyen').error(
       `Could not call payment/details for failed order ${order.orderNo}`,
     );
-    return handlePaymentError(order, options);
+    return handlePaymentError(order, adyenPaymentInstrument, options);
```

Comment:
this function contains three identical `handlePaymentError` calls. Can we reduce the amount of calls within this function? Calling it once would be cleaner.